spec/support/tempfile.cr

```diff
@@ -0,0 +1,43 @@
+require "tempfile"
+require "file_utils"
+
+SPEC_TEMPFILE_PATH = File.join(Tempfile.dirname, "crystal-spec-#{Random.new.hex(8)}")
+
+SPEC_TEMPFILE_CLEANUP = ENV["SPEC_TEMPFILE_CLEANUP"]? != "0"
+
+# Expands *paths* in a unique temp folder and yield them to the block.
+#
+# The *paths* are interpreted relative to a unique folder for every spec run and
+# prefixed by the name of the spec file that requests them.
+#
+# The constructed path is yielded to the block and cleaned up afterwards.
+#
+# Paths should still be uniquely chosen inside a spec file. This helper
+# ensures they're placed in the temporary location (`Tempfile.dirname`),
+# avoids name clashes between parallel spec runs and cleans up afterwards.
+#
+# The unique directory for the spec run is removed `at_exit`.
+#
+# If the environment variable `SPEC_TEMPFILE_CLEANUP` is set to `0`, no paths
+# will be cleaned up, enabling easier debugging.
+def with_tempfile(*paths, file = __FILE__)
+  calling_spec = File.basename(file).gsub("_spec.cr", "")
```

Comment:
You can use `delete("_spec.cr")` instead of `.gsub("_spec.cr", "")`

Reply:
@bew That's incorrect. `String#delete` works in such way only with `Char` argument.

Reply:
Are you sure about that? https://carc.in/#/r/47pg

Reply:
Yes, see [relevant `String#delete` docs](https://crystal-lang.org/api/0.24.2/String.html#delete%28%2Asets%29-instance-method).

Reply:
Oh that's really weird and not intuitive at all

Reply:
In this case `String#rchop` can be used, because the prefix to be removed is at the end.